### ISSUE TITLE
jdl: error passing `0` value in field annotations.

### ISSUE THREAD

**Issue by dharmeshmp** [jhipster/generator-jhipster#28312]
### **Issue Description**  
When adding new fields with `@defaultValue` in a JHipster entity, the generated Liquibase changelog does not correctly set the default values in the database. Specifically, the changelog generates the following incorrect XML:

```xml
<column name="games_played" type="integer" defaultValueNumeric="true">
    <constraints nullable="true" />
</column>
```

This leads to the following problems:
1. `defaultValueNumeric="true"` is invalid and does not set the intended default value (e.g., `0`).
2. The column constraints incorrectly allow `NULL` values, even though the `@defaultValue` annotation implies it should be `NOT NULL`.

---

### **Steps to Reproduce**  
1. Create a JHipster entity with fields that have `@defaultValue` annotations:
   ```java
   entity Player {
       displayName String required
       coins Integer required
       @defaultValue(0) gamesPlayed Integer
       @defaultValue(0) gamesWon Integer
   }
   ```
2. Generate the Liquibase changelog by running:
   ```bash
   jhipster jdl jhipster-jdl.jdl 
   ```
3. Check the generated Liquibase changelog file.

---

### **Expected Behavior**  
The generated changelog should correctly define default values and nullable constraints. For example:  
```xml
<column name="games_played" type="integer" defaultValueNumeric="0">
    <constraints nullable="false" />
</column>
```

---

### **Observed Behavior**  
The generated changelog incorrectly defines the columns as follows:  
```xml
<column name="games_played" type="integer" defaultValueNumeric="true">
    <constraints nullable="true" />
</column>
```

---

### **Environment**  
- **JHipster Version**: 8.8.0
- **JDK Version**: 17.0.12  
- **Database**: MySQL  
- **Operating System**: macOS  

---

### **Additional Context**  
This issue can lead to runtime errors or misaligned database schemas, especially when default values and non-null constraints are crucial for business logic.

---

### **Proposed Solution**  
Update the generator to:  
1. Ensure `defaultValueNumeric` is set to the numeric value provided in the `@defaultValue` annotation.  
2. Automatically set `nullable="false"` if a `@defaultValue` is defined.

**Comment by mshima** [comment]
The bug is in JDL parsing.
`@defaultValue(1)` works as expected, `@defaultValue(0)` fails.

**Comment by mshima** [comment]
I not sure if `nullable` behavior is expected.
@OmarHawk can you help here?

**Comment by OmarHawk** [comment]
> I not sure if `nullable` behavior is expected. @OmarHawk can you help here?

No, at the moment it is not expected that you don't pass a value. So, there should always be a parameter here.

**Comment by mshima** [comment]
Fixed by https://github.com/jhipster/generator-jhipster/pull/28753
